src/main/java/org/hibernate/models/internal/dynamic/DynamicAnnotationUsage.java

```
@@ -98,15 +110,76 @@ public <V> V setAttributeValue(String name, V value) {
 			values = new HashMap<>();
 		}
 
+		if ( value != null ) {
+
+			//noinspection unchecked
+			return (V) values.put( name, value );
+		}
 		//noinspection unchecked
-		return (V) values.put( name, value );
+		return (V) values;
```

I don't think this change is correct (casting the map of values to a single value type?), we should just leave it as it was.

If we really want to validate the provided value is not `null`, we could throw an `IllegalArgumentException`.

---

it seems that `null` is not supposed to be present has a value in fact what was happening before for a null value we were looking for the default value, what i noticed is that we set null overriding the default value

---

probably we should raise an `IllegalArgumentException`

---

Ok, than the check is fine but `return (V) values;` is not - maybe `return (V) values.get( name );`?

---

value should not be allowed to be null.  That condition should throw an exception.

---

done :)